File: 0257-binary-tree-paths/solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<String> binaryTreePaths(TreeNode root) {
        List<String> ansList = new ArrayList<String>();
        String path = "";

        if(root == null) return ansList;

        helper(root, ansList, path);
        return ansList;
    }

    public void helper(TreeNode root, List<String> ansList, String path){
        if(root == null) return ;
        if(root.left == null && root.right == null){
            path+=(root.val);
            ansList.add(path);
            return ;
        }

        helper(root.left,  ansList, path + root.val + "->");
        helper(root.right, ansList, path + root.val + "->");


    }
}

File: 0257-binary-tree-paths/solution.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
    void helper(TreeNode* root, vector<string> &ans, string temp) {
        if (root == nullptr) return;

        // Add current node value
        temp += to_string(root->val);

        // If leaf node, store path
        if (root->left == nullptr && root->right == nullptr) {
            ans.push_back(temp);
            return;
        }

        // Continue path
        temp += "->";

        helper(root->left, ans, temp);
        helper(root->right, ans, temp);
    }

public:
    vector<string> binaryTreePaths(TreeNode* root) {
        vector<string> ans;
        if (root == nullptr) return ans;

        helper(root, ans, "");
        return ans;
    }
};
